pkg/frontend/querymiddleware/limits.go

```diff
@@ -216,8 +217,8 @@ func newLimitedParallelismRoundTripper(next http.RoundTripper, codec Codec, limi
 }
 
 func (rt limitedParallelismRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
-	ctx, cancel := context.WithCancel(r.Context())
-	defer cancel()
+	ctx, cancel := context.WithCancelCause(r.Context())
+	defer cancel(cancellation.NewErrorf("limitedParallelismRoundTripper finished"))
```

Comment:
same comment here about rewording this so it's slightly less cryptic: how about "executing parallel subqueries finished"?

Reply:
Just like in the other one, this message shouldn't reach the customer.

Reply:
I've changed this to `executing parallel queries finished` in db107b9.